workers/notifier/src/main/kotlin/notifier/NotifierRunner.kt

```diff
@@ -70,17 +72,36 @@ class NotifierRunner {
                 workerContext.configManager.getSecret(Path(it))
             }
 
-            val notifierConfiguration = if (mailServerUser != null && mailServerPassword != null) {
-                NotifierConfiguration(
-                    mail = config.mail?.mailServerConfiguration?.copy(
+            val sendMailConfiguration = if (mailServerUser != null && mailServerPassword != null) {
+                config.mail?.mailServerConfiguration?.copy(
                         username = mailServerUser,
                         password = mailServerPassword
                     )?.mapToOrt()
-                )
             } else {
-                NotifierConfiguration()
+                SendMailConfiguration()
+            }
+
+            val jiraRestClientUsername = config.jira?.jiraRestClientConfiguration?.username?.let {
+                workerContext.configManager.getSecret(Path(it))
+            }
+            val jiraRestClientPassword = config.jira?.jiraRestClientConfiguration?.password?.let {
+                workerContext.configManager.getSecret(Path(it))
+            }
+
+            val jiraConfiguration = if (jiraRestClientUsername != null && jiraRestClientPassword != null) {
+                config.jira?.jiraRestClientConfiguration?.copy(
+                        username = jiraRestClientUsername,
+                        password = jiraRestClientPassword
+                    )?.mapToOrt()
+            } else {
+                JiraConfiguration()
             }
 
+            val notifierConfiguration = NotifierConfiguration(
+                mail = sendMailConfiguration,
+                jira = jiraConfiguration
```

Comment:
IMO we shouldn't use the default mail and Jira configurations, if the user didn't configure them.
I think both values should be `null` if they are not configured. This causes a different behavior in the ORT notifier, where `null` is handled separately:
https://github.com/oss-review-toolkit/ort/blob/2fcae578cfa3b5b6e451e7ab0300699c2b4d7982/notifier/src/main/kotlin/Notifier.kt#L46-L47

Reply:
Actually, we have no other choice than providing default Mail and Jira configurations, if the user did not explicitly configure one, because, as you already pointed out, the _Notifier_ from the oss-review-toolkit "injects" variables _mailClient_ and _jiraClient_ into the script _reporter-notification.kts_ (ort-server-config). If only **one** of these variables is not injected, then the complete kotlin script **will not compile**, and neither Mail nor Jira notifications are created at all.

_Unresolved reference: mailClient
Unresolved reference: jiraClient_

This raises the question, if there should be a refactoring in a way that there is one kotlin reporter script for Mail notifications, and another one for Jira notifications. Then, depending on which configurations are present, one or the other script could be executed or not. But IMO this would be a separate explicit change request.

Reply:
> If only one of these variables is not injected, then the complete kotlin script will not compile

Actually, I believe these tests prove the contrary:

[Run a script without a Jira configuration](https://github.com/oss-review-toolkit/ort/blob/2fcae578cfa3b5b6e451e7ab0300699c2b4d7982/notifier/src/test/kotlin/NotifierTest.kt#L41-L49)

[Run a script without a mail configuration](https://github.com/oss-review-toolkit/ort/blob/2fcae578cfa3b5b6e451e7ab0300699c2b4d7982/notifier/src/test/kotlin/NotifierTest.kt#L61-L69)

Reply:
You convinced me. Let's stick to the contract. If a user provides a notifier script that contains notification code for both mail and jira, but forgot to take care for proper of mail and jira, then better let the whole script crash.